logicrunner/executionbroker.go

```diff
@@ -243,11 +256,39 @@ func (q *ExecutionBroker) processTranscript(ctx context.Context, transcript *com
 		"request": transcript.RequestRef.String(),
 		"broker":  q.name,
 	})
-	logger.Debug("processed request")
 
-	replyData, err := q.requestsExecutor.ExecuteAndSave(ctx, transcript)
-	if err != nil {
-		logger.Warn("contract execution error: ", err)
+	var (
+		replyData insolar.Reply
+		err       error
+	)
+
+	if transcript.Request.CallType == record.CTMethod {
+		logger.Info("processing transcript with method")
+		var objDesc artifacts.ObjectDescriptor
+		objDesc, err = q.artifactsManager.GetObject(ctx, *transcript.Request.Object, &transcript.RequestRef)
+		if err == nil {
+			transcript.ObjectDescriptor = objDesc
+
+			if !transcript.Request.Immutable &&
+				transcript.ObjectDescriptor.EarliestRequestID() != nil &&
+				!transcript.RequestRef.GetLocal().Equal(*transcript.ObjectDescriptor.EarliestRequestID()) {
+				logger.Info("Got different earliest request from ledger")
+
+				q.cleanMutableQueue(ctx)
+				q.ledgerHasMoreRequests = true
```

Comment:
this is wrong. ledgerHasMoreRequests should be changed under lock. move it into cleanMutableQueue

Reply:
forgot about it, will fix it right away